programs/Makefile

```diff
@@ -67,7 +67,10 @@ endif
 
 # zlib detection
 VOID = /dev/null
-HAVE_ZLIB := $(shell echo "\#include <zlib.h>\nint main(){}" | $(CC) -o $(VOID) -x c - -lz 2> $(VOID) && echo 1 || echo 0)
+HAVE_ZLIB := $(shell echo -e "\#include <zlib.h>\nint main(){}" | $(CC) -o have_zlib -x c - -lz 2> $(VOID) && echo 1 || echo 0)
+ifeq ($(HAVE_ZLIB), 1)
```

Comment:
> echo -e isn't available on all platforms, including my Mac, but I'm not sure what it can be replaced with.

What shell are you using? With Linux and MSYS `-e` is required and can be selected as beginning of a script:
`playTests.sh:1` `#!/bin/sh -e`
It seems to work with macOS. Maybe `-e` is ignored.